### ISSUE TITLE
outputting maps not working

### ISSUE THREAD

**Issue by rbuels** [hashicorp/terraform#8384]
### Terraform Version
0.7.0

### Affected Resource(s)
Please list the resources as a list, for example:
- output

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
output "test_map" {
    value = {
        foo = "bar"
    }
}
```

### Debug Output
https://gist.github.com/rbuels/2832e49a256199f857037fda3a2b58ed

### Expected Behavior
Terraform plan should at least have run without error

### Actual Behavior
Terraform plan emitted the error: `output test_map is not a valid type ([]map[string]interface {})`'

### Steps to Reproduce
1.  Make a file with a map output, as shown above.
2. Run terraform plan on it.

### References
Wasn't support for this supposed to be added by #5841?
https://github.com/hashicorp/terraform/pull/5841

**Comment by rcha86** [comment]
Also seeing this problem. What is the correct syntax/formatting for this?

**Comment by cornfeedhobo** [comment]
Same problem here

**Comment by rcha86** [comment]
After digging around some more, I figured out how to create maps in output. You have to use the map function. See [here](https://www.terraform.io/docs/configuration/interpolation.html#map_key_value_)

```
output "test-map" { 
  value = "${map(
    "test-value-1", "map-value-1"
    "test-value-2", "map-value-2"
  )}"
}
```

However, terraform still does not support nested maps. [Open issue](https://github.com/hashicorp/terraform/issues/2114)

**Comment by jbardin** [comment]
Closed via #9268
